### PR TITLE
[Reviewer: Ellie] Multiple DNS servers

### PR DESCRIPTION
**Pull request by rkd-msw**
Counterpart to https://github.com/Metaswitch/sprout/pull/955

### CODE REVIEW

File: include/dnscachedresolver.h [hunk 1]
```diff
@@ -177,7 +178,8 @@ class DnsCachedResolver
   void wait_for_replies(DnsChannel* channel);
   static void destroy_dns_channel(DnsChannel* channel);
 
-  struct in_addr _dns_server;
+  struct in_addr _dns_servers[3];
```
Comment (eleanor-merry) [inline include/dnscachedresolver.h]: Can max_size be defined and used here and in the constructor

File: src/dnscachedresolver.cpp [hunk 2]
```diff
@@ -83,6 +83,34 @@ DnsResult::~DnsResult()
...
+  _dns_servers_count = std::min(max_servers, dns_servers.size());
+
+  LOG_STATUS("Creating Cached Resolver using servers:");
+  for (size_t i = 0; i < _dns_servers_count; i++)
+  {
+    LOG_STATUS("    %s", dns_servers[i].c_str());
+    // Parse the DNS server's IP address.
+    if (!inet_aton(dns_servers[i].c_str(), &(_dns_servers[i])))
+    {
+      LOG_ERROR("Failed to parse '%s' as IP address - defaulting to 127.0.0.1", dns_servers[i].c_str());
```
Comment (eleanor-merry) [inline src/dnscachedresolver.cpp]: Will there be any issues if multiple DNS servers are set to 127.0.0.1?

**Comment by eleanor-merry** [comment]
Looks good

File: include/dnscachedresolver.h (hunk 1 above)
Comment (rkd-msw) [inline include/dnscachedresolver.h]: Not nicely - if I define DNSResolver::MAX_SERVERS that can't be used as an array size, and I don't really want to `#define` it.

File: src/dnscachedresolver.cpp (hunk 2 above)
Comment (rkd-msw) [inline src/dnscachedresolver.cpp]: Probably not, but I'll check.

**Pull request merged**